---
title: "Exploratory analysis of a GEO dataset"
author: "Marta Ibáñez Lligoña"
date: "5/21/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# PART 1
# Introduction

# Summary
The study is entitled as "Gene Expression data from human mesenchymal stromal cells"

The summary of the study says the following *Cellular therapy is proposed for tendinopathy treatment. Bone marrow- (BM-MSC) and adipose tissue- (ASC-) derived mesenchymal stromal cells are candidate populations for such a therapy.We used microarrays to evaluate the basal gene expression in human BM-MSCs and ASCs and to create list of differentially expressed genes between analyzed groups.*

### Experimental Design
The microarray expression analysis was performed on RNA isolated from hASC (n=5) and hBM-MSC cells (n=6) using the Affymetrix Gene Atlas system.

### Citations
The citations are missing.

# Data description
There are two files: the target files and the expression matrix.

```{r}
expression <- as.matrix(read.delim("~/Desktop/Universitat/2_ANY/3_TRIMESTRE/Omics Techniques/Ibanez_Marta-OmicsTechniques/Exercise_1/expression.txt", row.names=1, comment.char="#"))
targets <- read.delim("~/Desktop/Universitat/2_ANY/3_TRIMESTRE/Omics Techniques/Ibanez_Marta-OmicsTechniques/Exercise_1/targets.txt", header=TRUE)
```



```{r}
show(targets)
dim(expression)
summary(expression)
mypatient <- paste0("Patient", 1:11)
```

We can now check the type of files they are:
```{r}
class(targets)
class(expression)
```
The data is a matrix and contains 11 different columns, which I assume they are the different samples.

```{r}
boxplot(expression, las=2)
```

Now, I create my own information:

```{r}
myInfo = list(myName= "Marta Ibáñez", myLab = "Bioinformatics Lab", myContact = "marta.ibanez@alum.esci.upf.edu", myTitle = "Practical Exercise on ExpressionSets")
show(myInfo)
```


# PART 2, Bionconductor classes for omics data
```{r}
require(Biobase)
```

```{r}

myEset <- ExpressionSet(expression)
class(myEset)
show(myEset)
```

## Information about covariates
```{r}
columnDesc <- data.frame(labelDescription = c("Sample Names", "Patients", "Groups"))
myAnnotDF <- new("AnnotatedDataFrame", data = targets, varMetadata= columnDesc)
show(myAnnotDF)
````

```{r}
phenoData(myEset) <- myAnnotDF
rownames(pData(myAnnotDF)) <- pData(myAnnotDF)$SampleName
myEset <- ExpressionSet(assayData = expression, phenoData = myAnnotDF)
show(myEset)
```

## Storing information about the experiment

```{r}
myDesc <- new("MIAME", name= myInfo[["myName"]], lab= myInfo[["myLab"]], contact = myInfo[["myContact"]], title = myInfo[["myTitle"]])
print(myDesc)
```


## Adding information about features
```{r}
myEset <- ExpressionSet(assayData = expression, fetureNames = mypatient, phenoData = myAnnotDF, experimentData = myDesc)
show(myEset)
```

## Using objects of class ExpressionSet
```{r}
dim(exprs(myEset))
class(phenoData(myEset))
class(pData(phenoData(myEset)))
head(pData(phenoData(myEset)))
head(pData(myEset))
```

Then, I tried subsetting the expression set:
```{r}
smallEset <- myEset[1:15, c(1:3,6:11)]
dim(exprs(smallEset))
dim(pData(smallEset))
head(pData(smallEset))
all(colnames(exprs(smallEset))==rownames(pData(smallEset)))
```


# Working with GEOquery

```{r}
require(GEOquery)
gse <- getGEO("GSE128949")
class(gse)
names(gse)
gse[[1]]
esetFromGEO<-gse[[1]]
```







